packages/web/src/components/PopupModal/utils.ts

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+interface PopupProps {
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  onConfirm?: () => Promise<void>;
+  setModalHeader: React.Dispatch<React.SetStateAction<string>>;
+  setBodyText: React.Dispatch<React.SetStateAction<string | undefined>>;
+  onClose: () => void;
+  setConfirmButtonVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  onOpen: () => void;
+  isLoading: boolean;
+  succussMessage?: string;
+  errorMessage?: string;
+  openButtonPropsLocal: any;
+  header: string;
+}
+
+export const popupProps = ({
+  setIsLoading,
+  onConfirm,
+  setModalHeader,
+  setBodyText,
+  onClose,
+  setConfirmButtonVisible,
+  onOpen,
+  isLoading,
+  succussMessage,
+  errorMessage,
+  openButtonPropsLocal,
+  header,
+}: PopupProps) => {
+  const onConfirmClick = async () => {
+    try {
+      setIsLoading(true);
+      await onConfirm!();
```

Comment:
We can't assert this exists; if it needs to exist, change the types to make it required instead of optional
```suggestion
      await onConfirm();
```